components/common/comment_icon.tsx

```diff
@@ -11,7 +11,16 @@ import {localizeMessage} from 'utils/utils.jsx';
 
 import ReplyIcon from 'components/widgets/icons/reply_icon';
 
-export default class CommentIcon extends React.PureComponent {
+type Props = {
+    extraClass: string;
+    postId: string;
+    location: string;
```

Comment:
You are allowing any string as location, and before was only or `Locations.CENTER` or `Locations.SEARCH`.

Reply:
For that, I'll have to convert `utils/constants.jsx` to typescript because `Locations.CENTER` is a value(read variable) and cannot be used as a type. It needs to be converted to an `enum`.

Should I convert just the location part or leave it for another PR?

Reply:
I think we can convert the `constants.jsx later`, but for now I would use something like `location: 'CENTER' | 'SEARCH';`

Reply:
I already tried that but it didn't work because `'CENTER'` is not a type.

Reply:
you can define types based on string literals, see for example here: https://github.com/mattermost/mattermost-webapp/blob/master/components/loading_screen.tsx#L8

Reply:
This is what I'm talking about:
```
components/common/comment_icon.tsx:27:9 - error TS2322: Type 'string' is not assignable to type '"CENTER" | undefined'.

27         location: Locations.CENTER,
```
I couldn't find any way to use `"CENTER"` as type and keep `location: Locations.CENTER`

Reply:
Probably you can use directly `"CENTER"` there, that is de default value, and its ok, the constants are great, but in this case the type information is good enough for that.

Reply:
I mean `location: "CENTER"`